From b2f80154ab9eff10fd730cebccf98db8127adf15 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Fri, 22 Jan 2016 19:22:55 -0800
Subject: [PATCH 1120/2848] HDFS-9350. Avoid creating temprorary strings in
 Block.toString() and getBlockName() (Staffan
 Friberg via cmccabe)

(cherry picked from commit e63388fdf22b5fd20ca00f9fad9f40656f117d95)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java

Change-Id: I9b996cceead12e51a79ff4d9c1374a02849f4e74
---
 .../org/apache/hadoop/hdfs/protocol/Block.java     |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
index b35365a..f86854d 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/protocol/Block.java
@@ -133,7 +133,8 @@ public void setBlockId(long bid) {
   /**
    */
   public String getBlockName() {
-    return BLOCK_FILE_PREFIX + String.valueOf(blockId);
+    return new StringBuilder().append(BLOCK_FILE_PREFIX)
+        .append(blockId).toString();
   }
 
   /**
@@ -157,7 +158,9 @@ public void setGenerationStamp(long stamp) {
    */
   @Override
   public String toString() {
-    return getBlockName() + "_" + getGenerationStamp();
+    StringBuilder sb = new StringBuilder();
+    this.appendStringTo(sb);
+    return sb.toString();
   }
 
   public void appendStringTo(StringBuilder sb) {
-- 
1.7.9.5

